lalrpop/src/api/test.rs

```diff
@@ -0,0 +1,168 @@
+use std::env::{current_dir, set_current_dir, set_var, temp_dir};
+use std::fs;
+use std::path;
+use std::sync::{LockResult, Mutex, MutexGuard};
+
+use super::*;
+
+// tests may be run in parallel and these tests affect global state, so lock
+static API_TEST_MUTEX: Mutex<i32> = Mutex::new(0);
+
+const TEST_DIR: &str = "lalrpop-test";
+const CUSTOM_TEST_DIR: &str = "lalrpop-test2";
+
+#[derive(Debug, PartialEq)]
+enum GenFileLoc {
+    Src,
+    Other,
+    Root,
+    OutDir,
+    CustomOut,
+    DoesntExist,
+}
+
+// Set up for API tests.  The directory structure in test_files
+// is:
+//
+// outer.lalrpop
+// other
+//   - other.lalrpop
+// src
+//   - src.lalrpop
+//
+// So we want to set CWD to directly above that, and OUT_DIR to a temp directory
+fn setup() -> (path::PathBuf, LockResult<MutexGuard<'static, i32>>) {
+    let lock = API_TEST_MUTEX.lock();
+    let orig_dir = current_dir().unwrap();
+    set_current_dir(path::Path::new("./src/api/test_files")).unwrap();
```

Comment:
I did a quick double check and Rust does do the right thing here for Windows https://doc.rust-lang.org/std/path/struct.Path.html